mesheryctl/internal/cli/root/filter/list.go

```diff
@@ -92,57 +99,110 @@ mesheryctl filter list 'Test Filter' (maximum 25 filters)
+		} else {
 			for _, v := range response.Filters {
 				FilterID := utils.TruncateID(v.ID.String())
 				var UserID string
 				if v.UserID != nil {
-					UserID = *v.UserID
+					UserID = utils.TruncateID(*v.UserID)
 				} else {
 					UserID = "null"
 				}
 				FilterName := v.Name
-				CreatedAt := fmt.Sprintf("%d-%d-%d %d:%d:%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year(), v.CreatedAt.Hour(), v.CreatedAt.Minute(), v.CreatedAt.Second())
-				UpdatedAt := fmt.Sprintf("%d-%d-%d %d:%d:%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year(), v.UpdatedAt.Hour(), v.UpdatedAt.Minute(), v.UpdatedAt.Second())
+				CreatedAt := fmt.Sprintf("%d-%d-%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year())
+				UpdatedAt := fmt.Sprintf("%d-%d-%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year())
 				data = append(data, []string{FilterID, UserID, FilterName, CreatedAt, UpdatedAt})
 			}
-			utils.PrintToTableWithFooter([]string{"FILTER ID", "USER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", "", ""})
-			return nil
+			header = []string{"FILTER ID", "USER ID", "NAME", "CREATED", "UPDATED"}
+			footer = []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", "", ""}
 		}
 
-		// Check if meshery provider is set
-		if provider == "None" {
-			for _, v := range response.Filters {
-				FilterName := strings.Trim(v.Name, filepath.Ext(v.Name))
-				FilterID := utils.TruncateID(v.ID.String())
-				CreatedAt := fmt.Sprintf("%d-%d-%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year())
-				UpdatedAt := fmt.Sprintf("%d-%d-%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year())
-				data = append(data, []string{FilterID, FilterName, CreatedAt, UpdatedAt})
-			}
-			utils.PrintToTableWithFooter([]string{"FILTER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", ""})
+		if cmd.Flags().Changed("page") {
+			utils.PrintToTableWithFooter(header, data, footer)
 			return nil
 		}
-		for _, v := range response.Filters {
-			FilterID := utils.TruncateID(v.ID.String())
-			var UserID string
-			if v.UserID != nil {
-				UserID = utils.TruncateID(*v.UserID)
-			} else {
-				UserID = "null"
-			}
-			FilterName := v.Name
-			CreatedAt := fmt.Sprintf("%d-%d-%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year())
-			UpdatedAt := fmt.Sprintf("%d-%d-%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year())
-			data = append(data, []string{FilterID, UserID, FilterName, CreatedAt, UpdatedAt})
-		}
-		utils.PrintToTableWithFooter([]string{"FILTER ID", "USER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", "", ""})
-		return nil
+		return handlePagination(data, header, footer)
 	},
 }
 
+func handlePagination(data [][]string, header, footer []string) error {
```

Comment:
Thank you for encapsulating this logic into a function. Please consider moving it to our `utils` module and refactoring it into a generic function. We're currently using the same logic for pagination in all `mesheryctl commands`